src/ansys/mechanical/core/embedding/app.py

```diff
@@ -437,6 +439,14 @@ def project_directory(self):
         """Returns the current project directory."""
         return self.DataModel.Project.ProjectDirectory
 
+    @property
+    def messages(self):
```

Comment:
I don't like that `_update_messages_cache()` has to be done explicitly. The message manager should be responsible for keeping itself up to date. Use an event handler